src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1802,29 +1809,29 @@ private BetterResult BetterConversionFromExpression(BoundExpression node, TypeSy
 
             var lambdaOpt = node as UnboundLambda;
 
-            // Given an implicit conversion C1 that converts from an expression E to a type T1, 
-            // and an implicit conversion C2 that converts from an expression E to a type T2,
-            // C1 is a better conversion than C2 if E does not exactly match T2 and one of the following holds:
             bool t1MatchesExactly = ExpressionMatchExactly(node, t1, ref useSiteDiagnostics);
             bool t2MatchesExactly = ExpressionMatchExactly(node, t2, ref useSiteDiagnostics);
 
             if (t1MatchesExactly)
             {
                 if (!t2MatchesExactly)
                 {
-                    // - E exactly matches T1
                     okToDowngradeToNeither = lambdaOpt != null && CanDowngradeConversionFromLambdaToNeither(BetterResult.Left, lambdaOpt, t1, t2, ref useSiteDiagnostics, false);
                     return BetterResult.Left;
                 }
+                else if (lambdaOpt == null)
+                {
+                    // both match exactly, non-lambda case.
+                    // Neither is a better conversion from expression
+                    return BetterResult.Neither;
+                }
```

Comment:
I am not comfortable with this change. It also doesn't match the latest specification quoted by Mads at [https://github.com/dotnet/roslyn/issues/14117](https://github.com/dotnet/roslyn/issues/14117)